src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,49 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name){
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  die(){
+    Animal.alive = Animal.alive.filter(animal => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name){
+    super(name);
+    this.hidden = false;
+  }
+
+  hide(){
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name){
```

Comment:
![image](https://github.com/user-attachments/assets/30897e11-6661-402c-b815-1844a4705d0a)

Fix it (you can completely remove constructor here - it should work properly anyway)